internal/controllers/addon/controller.go

```diff
@@ -244,14 +282,21 @@ func (r *AddonReconciler) Reconcile(
 			ctx, log.WithName("UpgradePolicyStatusReporter"), addon,
 		)
 
-		// Finally, update the Status back to the kube-api
-		// This is the only place where Status is being reported.
+		if err != nil {
+			return
+		}
```

Comment:
Thanks for adding this following `err = r.handleUpgradePolicyStatusReporting(`. 👍 

During this review i noticed that their is a `if err != nil` right above `r.handlerUpgradePolicyStatusReporting(` call. can we remove it ([this](https://github.com/openshift/addon-operator/pull/293/files#diff-356bd256e811b822d211e8f4826da4860bbddac5b5f761520767b785024c63fbR278))

Reply:
Why would we want to remove it? I agree that the way we are handling errors from talking to API's external to the cluster is not right. (We can fix that as part of a different issue?)

Reply:
it makes sense now. 🧑‍💻